### ISSUE TITLE
Application unavailable while deploy

### ISSUE THREAD

**Issue by bibimij** [Unitech/pm2-deploy#131]

**Comment by constantx** [comment]
I'm having the same issue: filed on https://github.com/Unitech/pm2/issues/2934

**Comment by constantx** [comment]
I wonder if this is still being maintained, cause that solution is around since 2014…

**Comment by constantx** [comment]
@Unitech ping

**Comment by bibimij** [comment]
@constantx I feel your pain, bro. If nobody worried since 2014 and consider it normal behaviour, I've got a bad news for all frontend community.

**Comment by vmarchaud** [comment]
First you have a different use case the one that was intended when first developing the tool, please keep in mind that.
Secondly like any open-source project you are free to contribute back and implement another deployment flow.
Thirdly i started doing coding [deployerjs](https://github.com/vmarchaud/deployerjs) so because it got the same thought as you and `pm2-deploy` isn't really designed to have different deployment strategy. But it take times to develop things like this specially when you are doing this on your free time.

Conclusion : Indeed `pm2-deploy` isn't maintained because the wanted behavior of the flow is working, but it's pretty much impossible to implement other behavior and there isn't easy solution.

**Comment by bibimij** [comment]
@vmarchaud of course, open source owes me nothing. But before the start I would like to be sure: is it bug or feature? So, if it's feature, my work will be vain.

**Comment by vmarchaud** [comment]
@bibimij It's both, at the start it was a feature because it's was designed like this, now it's more like a bug because it's blocking for people.

**Comment by bibimij** [comment]
@vmarchaud ok! I'll take a look on deployerjs. It isn't production solution and I can use it at my own risk, is it?

**Comment by vmarchaud** [comment]
@bibimij It's still in development, don't remember if the default strategy (the same one as `pm2 deploy` via git) is working
